jupyter/utility/latest runs from spreadsheet analysis.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+
+pd.set_option('display.max_rows', 500)
+pd.set_option('display.max_columns', 500)
+pd.set_option('display.width', 1000)
+
+
+# In[3]:
+
+
+## reading exported csv
+
+# to get csv - save 'BEAM Deploy Status and Run Data' as csv
+# if there is not enough permissions - save a copy and then save as csv
+
+data = pd.read_csv("../../../beam-production/jupyter/local_files/Copy of BEAM Deploy Status and Run Data - BEAM Instances.csv", parse_dates=['Time'])
+
+# using only runs from specific data 
+min_time = pd.to_datetime("2022-03-01") # yyyy-mm-dd
+data = data[data['Time'] > min_time].copy()
+
+print(f"there are roughly {len(data) / 2} runs since {min_time}")
+print(f"the latest run is from {data['Time'].max()}")
+
+data['Month Period'] = data['Time'].dt.strftime('%Y-%m')
+print(f"following data periods are included: {sorted(data['Month Period'].unique())}")
+
+data.head(3)
+
+
+# In[4]:
+
+
+## getting data frame with each row as one simulation
+
+take_first_columns = ['Run Name','Month Period','Branch','Instance type']
+
+df = data.groupby("Host name").agg(list)
+for col in take_first_columns:
+    df[col] = df.apply(lambda r: r[col][0], axis=1)
+
+df['Time Start'] = df.apply(lambda r: r['Time'][0], axis=1)
+df['Time Stop'] = df.apply(lambda r: r['Time'][-1], axis=1)
+
+all_columns = set(df.columns)
+taken_columns = take_first_columns + ['Time Start', 'Time Stop']
+
+df = df[taken_columns].copy()
+
+print(f"removed columns: {list(sorted(all_columns - set(taken_columns)))}")
+
+# fix for some wierd shift in the spreadsheet for few rows
+for v in ['ec2-18-221-208-40.us-east-2.compute.amazonaws.com','ec2-3-144-69-95.us-east-2.compute.amazonaws.com','ec2-52-15-53-101.us-east-2.compute.amazonaws.com']:
+    df.replace(to_replace=v, value='r5d.24xlarge', inplace=True)
+
+df['duration_hours'] = (df['Time Stop'] - df['Time Start']).astype('timedelta64[h]')
+
+df.head(3)
+
+
+# In[5]:
+
+
+## just a check
+df['Instance type'].unique()
+
+
+# In[6]:
+
+
+## calculating a price in USD of each simulation
+
+instance_to_price = {'r5d.24xlarge':6.912, 
+                     'm5d.24xlarge':5.424, 
+                     'r5.xlarge':0.252, 
+                     'r5.24xlarge':6.048,
+                     'r5.8xlarge':2.016, 
+                     'm5.24xlarge':4.608, 
+                     'r5.2xlarge':0.504, 
+                     'm4.16xlarge':3.20
+                    }
+
+missing_instance_types = set()
+def get_price(row):
```

Comment:
Maybe we could extract this and other utility method to a python package? Then we could use it here and in the jupyter notebook.